content/en/docs/pluggable-components/logging.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Enable Logging"
+keywords: "Kubernetes, Elasticsearch, KubeSphere, Logging, logs"
+description: "How to Enable KubeSphere Logging System"
+
+linkTitle: "Enable Logging"
+weight: 3535
+---
+
+## What is KubeSphere Logging System
+
+KubeSphere provides a powerful, holistic and easy-to-use logging system for log collection, query and management. It covers logs from at varied levels, including tenants, infrastructure resources, and applications. Users can search logs from different dimensions, such as project, workload, Pod and keyword. Compared with Kibana, the tenant-based logging system of KubeSphere features better isolation and security among tenants as each tenant can only view his or her own logs. Apart from KubeSphere's own logging system, the container platform also allows users to add third-party log collectors, such as Elasticsearch, Kafka and Fluentd. 
+
+For more information, see Logging, Events and Auditing.
+
+{{< notice note >}}
+
+Logging needs to be enabled first if you need to install Events or Auditing.
+
+{{</ notice >}} 
+
+## Enable Logging before Installation
+
+### Installing on Linux
+
+When you install KubeSphere on Linux, you need to create a configuration file, which lists all KubeSphere components.
+
+1. In the tutorial of [Installing KubeSphere on Linux](https://kubesphere-v3.netlify.app/docs/installing-on-linux/introduction/multioverview/), you create a default file **config-sample.yaml**. Modify the file by executing the following command:
+
+```bash
+vi config-sample.yaml
+```
+
+{{< notice note >}}
+
+If you adopt [All-in-one Installation](https://kubesphere-v3.netlify.app/docs/quick-start/all-in-one-on-linux/), you do not need to create a config-sample.yaml file as you can create a cluster directly. Generally, the all-in-one mode is for users who are new to KubeSphere and look to get familiar with the system. If you want to enable Logging in this mode (e.g. for testing purpose), refer to the following section to see how Logging can be installed after installation.
+
+{{</ notice >}}
+
```

Comment:
Please add a notice for using the external Elasticsearch in the `config-example.yaml` if users are choosing to install a production environment (the external ES is not necessary for development or testing), see this [guide](https://kubesphere.com.cn/docs/installation/install-logging/#enable-logging-system-before-installation) for details.

```
es:  # Storage backend for logging, tracing, events and auditing.
      elasticsearchMasterReplicas: 1   # total number of master nodes, it's not allowed to use even number
      elasticsearchDataReplicas: 1     # total number of data nodes
      elasticsearchMasterVolumeSize: 4Gi   # Volume size of Elasticsearch master nodes
      elasticsearchDataVolumeSize: 20Gi    # Volume size of Elasticsearch data nodes
      logMaxAge: 7                     # Log retention time in built-in Elasticsearch, it is 7 days by default.
      elkPrefix: logstash              # The string making up index names. The index name will be formatted as ks-<elk_prefix>-log
      externalElasticsearchUrl:   # The URL of external Elasticsearch
      externalElasticsearchPort: # The port of external Elasticsearch
```

Reply:
For Logging, Events, and Auditing system, as soon as users enable any of the three components, the installer will install (or rely on) an Elasticsearch. The notice above should be also added in the other two guides (KubeSphere Events System and KubeSphere Auditing System).